state-chain/custom-rpc/src/lib.rs

```diff
@@ -1277,6 +1280,24 @@ where
 			.cf_witness_count(self.unwrap_or_best(at), pallet_cf_witnesser::CallHash(hash.into()))
 			.map_err(to_rpc_error)
 	}
+
+	fn cf_get_events(&self, at: Option<state_chain_runtime::Hash>) -> RpcResult<Vec<String>> {
+		let decoder = state_chain_runtime::event_encoder::RuntimeEventDecoder::new();
+		let events = self
+			.client
+			.runtime_api()
+			.cf_block_events(self.unwrap_or_best(at))
+			.map_err(to_rpc_error)?;
+
+		Ok(events
+			.into_iter()
+			.map(|event| {
+				decoder.decode_event_to_json(event).unwrap_or_else(|err| {
+					format!("Failed to decode RuntimeError. Error: {:?}", err)
```

Comment:
We don't want to return an error inside the results. 

We could instead log a warning and insert a custom value with the hex-encoded raw event data, for example `{"name":"UnknownEvent","values":["0xabcd.."]}`.

Maybe it works if we add another `struct UnknownEvent(Vec<u8>)` wrapper type into the metadata. And if the RuntimeEvent fails to decode, we fall back to that.

Reply:
Got the UnknownEvent - but I didn't go through the metadata - and constructed the Value directly. i thought it would be easier this way